apps/ehr/test/e2e/page/StateDetailsPage.ts

```diff
@@ -1,6 +1,65 @@
 import { expect, Page } from '@playwright/test';
+import { dataTestIds } from '../../../src/constants/data-test-ids';
 
-export async function expectStateDetailsPage(state: string, page: Page): Promise<void> {
+export class StateDetailsPage {
+  #page: Page;
+
+  constructor(page: Page) {
+    this.#page = page;
+  }
+
+  async clickCancelButton(): Promise<void> {
+    await this.#page.getByTestId(dataTestIds.editState.cancelButton).click();
+  }
+
+  async clickSaveChangesButton(): Promise<void> {
+    await this.#page.getByTestId(dataTestIds.editState.saveChangesButton).click();
+    await this.#page.waitForTimeout(5000);
+  }
+
+  async verifyStateNameTitle(stateNameTitle: string): Promise<void> {
+    await expect(this.#page.getByTestId(dataTestIds.editState.stateNameTitle)).toHaveText(stateNameTitle);
+  }
+
+  async verifyStateNameField(stateNameField: string): Promise<void> {
+    await expect(this.#page.getByTestId(dataTestIds.editState.stateNameField).locator('input')).toHaveValue(
+      stateNameField
+    );
+  }
+
+  async setToggleOff(): Promise<void> {
+    await this.#page.getByTestId(dataTestIds.editState.operateInStateToggle).locator('input').setChecked(false);
+    await this.#page.waitForTimeout(5000);
+  }
+
+  async verifyToggleOff(): Promise<void> {
+    await expect(this.#page.getByTestId(dataTestIds.editState.operateInStateToggle).locator('input')).toBeChecked({
+      checked: false,
+    });
+  }
+
+  async setToggleOn(): Promise<void> {
+    await this.#page.getByTestId(dataTestIds.editState.operateInStateToggle).locator('input').setChecked(true);
+    await this.#page.waitForTimeout(5000);
```

Comment:
`waitForTimeout` in this PR increases execution time by 20 seconds. Let's find a way to verify the changes faster.